src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -199,13 +220,24 @@ impl<'a> TaskContext<'a> {
         coverage: BinaryCoverage,
         allowlist: TargetAllowList,
         heartbeat: Option<TaskHeartbeatClient>,
-    ) -> Self {
-        Self {
+        target_exe: String,
+    ) -> Result<Self> {
+        let cache = DebugInfoCache::new(allowlist.source_files.clone());
+        let loader = Loader::new();
+
+        let module: Box<dyn Module> = LoadModule::load(&loader, FilePath::new(target_exe)?)?;
+
+        cache.get_or_insert(&*module)?;
```

Comment:
By using `?` here, we bail on error. One type of error is that the module simply has no debuginfo. In almost all real life cases, this would signal an error in job submission. However, in general, we could imagine that the `target_exe` is a libfuzzer binary whose true target code is in some DLL that gets dynamically linked or loaded.

Question for the OneFuzz team to consider: is it a hard requirement that the `target_exe` include debuginfo?

I think it is probably fair to say yes, and it may even already be de facto true (I don't remember). But that should be understood and agreed upon.

Reply:
Regardless: consider adding a `context()` call here to note that this error is occurring in pre-caching, not during an iteration.